### DISCUSSION TITLE
How to set default search for React on page load?

### DISCUSSION THREAD

**Discussion by vonkoff** [algolia/instantsearch#5688] [Q&A]
I am not finding anything in react-hooks UI library on how to set a default search for people going to a certain route of my react app?

Anyone have a quick example of how to do it?

**Comment by Haroenv** [comment]
You can use routing to synchronise (part of) the search state to the URL. A guide on that is here: https://www.algolia.com/doc/guides/building-search-ui/going-further/routing-urls/react-hooks/

**Reply by vonkoff** [reply]
Got help! Here is an example. https://codesandbox.io/s/initialuistate-example-g4vhp4

More info- https://www.algolia.com/doc/api-reference/widgets/ui-state/react-hooks/
